build_runner/lib/src/asset/file_based.dart

```diff
@@ -50,7 +50,7 @@ class FileBasedAssetReader implements RunnerAssetReader {
     for (var file in files) {
       // TODO(jakemac): Where do these files come from???
       if (path.basename(file.path).startsWith('._')) continue;
-      yield _fileToAssetId(file, packageNode);
+      yield _fileToAssetId(file as File, packageNode);
```

Comment:
This one seems to work at runtime without much consequence so I moved the cast. It might start failing in the VM in the future but we can address it then.